### PR TITLE
[BUGFIX] Fixed persist to be considered for spark fluent datasource

### LINKED ISSUE #8511: Not able to set persist to False using Spark Execution Engine
Hey Guys, I was trying to migrate GX from 0.16.5 and am not being able to, cause apparently since GX 0.16.12 supposedly there was a fix for the persist parameter to work. The thing is that I wanted it to be False, but it seems the parameter is not taking effect.
Anyone facing similar problems?
I was following this guide [How to connect to in-memory data in a Spark dataframe | Great Expectations](https://docs.greatexpectations.io/docs/0.15.50/guides/connecting_to_your_data/in_memory/spark/), which seems to have been removed. Also tried with fluent approach and still failing.
Is the property persist being considered/passed? Shouldn’t it be one of the parameters in add_or_update_spark ?

More information: https://discourse.greatexpectations.io/t/not-able-to-set-persist-to-false-using-spark-execution-engine/1320?u=jose.correia

### PR DESCRIPTION
**Pull request by jmcorreia**
Fix persist parameter to be considered for Spark Datasources, so that people can pass it as False.
Enhanced test to cover the code changes.

Fixes #8511 .

### CODE REVIEW

File: tests/datasource/fluent/test_spark_datasource.py
Comment (Kilo59) [inline tests/datasource/fluent/test_spark_datasource.py]: This part of the test shouldn't be necessary. This is just testing the yaml serialization logic which is tested elsewhere.

File: tests/datasource/fluent/test_spark_datasource.py
Comment (Kilo59) [inline tests/datasource/fluent/test_spark_datasource.py]: I think this might fit better in the `test_spark_config_passed_to_execution_engine` test below.
Or maybe that test should be parametrized to include the `persist` key.

File: tests/datasource/fluent/test_spark_datasource.py
Comment (jmcorreia) [reply tests/datasource/fluent/test_spark_datasource.py]: The idea was basically to ensure that the persist to False is really there. 
But I will do the suggested changes.

File: tests/datasource/fluent/test_spark_datasource.py
Comment (jmcorreia) [reply tests/datasource/fluent/test_spark_datasource.py]: Done

**Review by Kilo59 — APPROVED** [review]
Thanks for this!

I don't know Spark that well but this seems reasonable.
I'd like to get some reviews from some other engineers before merging.

File: great_expectations/datasource/fluent/sources.pyi
Comment (NathanFarmer) [inline great_expectations/datasource/fluent/sources.pyi:339]: Can you also add this parameter to the `*_spark_filesystem`, `*_spark_dbfs`, `*_spark_s3`, `*_spark_gcs`, and `*_spark_abs` methods in this file?

**Review by NathanFarmer — COMMENTED** [review]
One request from me. Thank you for this contribution!

File: great_expectations/datasource/fluent/sources.pyi
Comment (jmcorreia) [reply great_expectations/datasource/fluent/sources.pyi:339]: Nice callout, applied.
Anyway, I have also tested without adding persist as a clear parameter and it also works, as we have a * there.

**Comment by jmcorreia** [comment]
@anthonyburdi , @NathanFarmer could you please let me know if you still something missing? Really need this to be able to upgrade great-expectations on my side :)
Btw, this fixes only fluent datasource (spark_datasource), it does not fix for people using the non-fluent version sparkdf_datasource. Should I try to apply the fix there as well? And if I do it, can I change the default of force_reuse_spark_context to True?

Or if you think it is better, I can also propose the rest in a separate PR.

**Pull request merged**
